test/test_noise_std.py

```diff
@@ -8,146 +8,157 @@
+    signal_noise_ratio = 2.0
 
     # First expe
     # ##########
     support_size = 2
 
-    X, y, beta, noise = multivariate_1D_simulation(
+    X, y, beta, noise = multivariate_simulation(
         n_samples=n_samples,
         n_features=n_features,
         support_size=support_size,
-        sigma=sigma,
+        rho=0.25,
+        signal_noise_ratio=signal_noise_ratio,
         seed=0,
     )
 
     # max_iter=1 to get a better coverage
     sigma_hat, _ = reid(X, y, tolerance=1e-3, max_iterance=1)
-    expected = sigma
-
-    assert_almost_equal(sigma_hat / expected, 1.0, decimal=0)
+    expected_sigma = support_size / signal_noise_ratio
+    error_relative = np.abs(sigma_hat - expected_sigma) / expected_sigma
+    assert error_relative < 0.3
 
     # Second expe
     # ###########
     support_size = 0
 
-    X, y, beta, noise = multivariate_1D_simulation(
+    X, y, beta, noise = multivariate_simulation(
         n_samples=n_samples,
         n_features=n_features,
         support_size=support_size,
-        sigma=sigma,
-        seed=1,
+        signal_noise_ratio=signal_noise_ratio,
+        seed=2,
     )
 
     sigma_hat, _ = reid(X, y)
-    expected = sigma
-
-    assert_almost_equal(sigma_hat / expected, 1.0, decimal=1)
+    expected_sigma = 1.0  # when there is no signal, the variance is 1.0
+    error_relative = np.abs(sigma_hat - expected_sigma) / expected_sigma
+    assert error_relative < 0.2
 
 
 def test_group_reid():
     """Estimating (temporal) noise covariance matrix in two scenarios.
     First scenario: no data structure and a support of size 2.
     Second scenario: no data structure and an empty support."""
 
-    n_samples = 30
-    n_features = 50
-    n_times = 10
-    sigma = 1.0
-    rho = 0.9
-    corr = toeplitz(np.geomspace(1, rho ** (n_times - 1), n_times))
-    cov = np.outer(sigma, sigma) * corr
+    n_samples = 100
+    n_features = 20
+    n_target = 50
+    signal_noise_ratio = 3.0
+    rho_serial = 0.9
 
     # First expe
     # ##########
     support_size = 2
-
-    X, Y, beta, noise = multivariate_temporal_simulation(
+    X, Y, beta, noise = multivariate_simulation(
         n_samples=n_samples,
         n_features=n_features,
-        n_times=n_times,
+        n_targets=n_target,
         support_size=support_size,
-        sigma=sigma,
-        rho_noise=rho,
+        signal_noise_ratio=signal_noise_ratio,
+        rho_serial=rho_serial,
+        rho=0.0,
+        seed=0,
+    )
+    corr = (
+        support_size
+        / signal_noise_ratio
+        * toeplitz(np.geomspace(1, rho_serial ** (n_target - 1), n_target))
     )
 
     # max_iter=1 to get a better coverage
     cov_hat, _ = reid(X, Y, multioutput=True, tolerance=1e-3, max_iterance=1)
-    error_ratio = cov_hat / cov
-
-    assert_almost_equal(np.max(error_ratio), 1.0, decimal=0)
-    assert_almost_equal(np.log(np.min(error_ratio)), 0.0, decimal=1)
+    error_relative = np.abs(cov_hat - corr) / corr
```

Comment:
I don't think that this is the right test.

Reply:
Yes, I forgot to put back coverage variable.